### PR TITLE
Update roadmap 2023

### PR DESCRIPTION
**Pull request by cp-sumi-k**

### CODE REVIEW

**Review by cp-priyanka-g — APPROVED** [review]

**Review by cp-jagruti-a — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,58 +1,72 @@
...
 
-![web roadmap](https://github.com/canopas/web-developer-roadmap/blob/master/images/title_image.jpeg)
+![web roadmap](https://github.com/canopas/web-developer-roadmap/blob/master/images/title_image.png)
 
-Web Developer Roadmap 2022 is a learning path to understand web development, including frontend, backend and cloud(AWS).
+Web Developer Roadmap 2023 is a learning path to understand web development, including frontend, backend and cloud(AWS).
 
 ## How to Learn Web development?
 
 In web development, there are mainly four ends.
```
Comment (nidhi-canopas) [inline README.md]: I think this can be `A web development can be divided into four different parts`

File: README.md [hunk 2]
```diff
@@ -1,58 +1,72 @@
...
 
-#### Practical 1.1
+### Practical 1.1 - Basic commands and Version control
+
+- Write a commands for following operations in terminal
+  - List all files with details in directory
+  - Give only read permission to any file
+  - Give all read and write permissions to any file
+  - Get self IP address
+  - See disk space usage
```
Comment (nidhi-canopas) [inline README.md]: See -> Observe

**Review by cp-mansi-d — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,58 +1,72 @@
...
+  - See previously executed commands
```
Comment (nidhi-canopas) [inline README.md]: View previously executed commands history

File: README.md [hunk 4]
```diff
@@ -1,58 +1,72 @@
...
+## Sprint 1 - Basic linux commands, Version control, Web technologies and coding conventions
 
-#### Practical 1.1
+### Practical 1.1 - Basic commands and Version control
+
+- Write a commands for following operations in terminal
+  - List all files with details in directory
+  - Give only read permission to any file
+  - Give all read and write permissions to any file
+  - Get self IP address
```
Comment (nidhi-canopas) [inline README.md]: Get IP address of own pc

File: README.md [hunk 5]
```diff
@@ -1,58 +1,72 @@
...
+- [Basic linux commands](https://www.digitalocean.com/community/tutorials/linux-commands)
+
 - [What Is Version Control?](https://about.gitlab.com/topics/version-control/)
 - How to use git
   - [Version control with git](https://www.udacity.com/course/version-control-with-git--ud123)
   - [Git: The Beginner's Guide to Understanding Core Version Control Concepts](https://www.freecodecamp.org/news/git-the-laymans-guide-to-understanding-the-core-concepts/)
   - [Git commands](https://dzone.com/articles/top-20-git-commands-with-examples)
 
-#### Practical 1.2
+### Practical 1.2 - Basic web technology with coding conventions
```
Comment (nidhi-canopas) [inline README.md]: technology -> technologies

File: README.md [hunk 6]
```diff
@@ -1,58 +1,72 @@
...
 
-- UI Design
-  - Design static UI given in the [link](https://github.com/canopas/web-developer-roadmap/blob/master/images/static.png)
-  - Design responsive UI given in the [link](https://www.w3schools.com/w3css/tryw3css_templates_food_blog.htm)
+- UI Design by following coding standards
```
Comment (nidhi-canopas) [inline README.md]: UI design with coding standards

File: README.md [hunk 7]
```diff
@@ -123,17 +141,18 @@ It also includes overview about cloud(AWS) and some information about server.
...
-- Create a Blog application with following requirements
+- Create a Blog application with following requirements using PHP
   - It should have two sides
     - 1. Admin
     - 2. User
   - Implement Register and login functionality for both user and admin
   - Admin can add posts ( post fields - title, description, created_date, author, category, image )
   - Admin can decide which user can see the post
-  - Added Posts will be visible on user side (latest posts first)
+  - Added Posts will be visible on user side (Order by latest first)
```
Comment (nidhi-canopas) [inline README.md]: I think we don't need to mention to use order by.

File: README.md [hunk 8]
```diff
@@ -105,15 +120,18 @@ It also includes overview about cloud(AWS) and some information about server.
...
 - Perform following queries in MySQL
 
   - Create a table named `students` with fields id, first_name, last_name, standard, percentage, interest, etc... and insert data into it
   - Create table `student_attendances` with fields id, created_at, presence/absence fields and insert data into it
+  - Create a table named `teachers` with fields id, first_name, last_name, subject, interests etc... and insert data into it
   - Prepare queries to find student's presence/absence on a particular day
   - Find total absence/presence of every student
   - Find absent students with a percentage lower than 70.
+  - Find a student who has higest presence
+  - Get all student's and teacher's first_name, last_name, full_name and interest.
```
Comment (nidhi-canopas) [inline README.md]: I think we need some complex query instead of a simple one.

**Review by cp-dharti-r — APPROVED** [review]

File: README.md [hunk 9]
```diff
@@ -190,9 +209,9 @@ It also includes overview about cloud(AWS) and some information about server.
 
 ## Sprint 3 - Golang
 
-#### Practical 3.1
+### Practical 3.1
 
-- Implement APIs for music application with [given requirements](https://github.com/canopas/web-developer-roadmap/blob/master/music_app.md)
+- Implement [Music API application](https://github.com/canopas/web-developer-roadmap/blob/master/music_app.md) with given requirements
```
Comment (nidhi-canopas) [inline README.md]: Will Music API application be readable?

File: music_app.md [hunk 10]
```diff
@@ -4,9 +4,19 @@
...
+
+### Technologies
+ - Gin framework
+ - Sqlx or Gorm
+ - JWT
+
+### Application will have two type of users :
+
+- Admin - Should have Read/Write access
+- User - Should have Read access only, except Favourite/Unfavourite
```
Comment (nidhi-canopas) [inline music_app.md]: except Favourite/Unfavourite functionality

**Review by nidhi-canopas — APPROVED** [review]

**Review by cp-dharti-r — APPROVED** [review]

**Pull request merged**